cli/internal/state/state.go

```diff
@@ -555,6 +557,23 @@ func (s *State) Constraints() []*validation.Constraint {
 	return []*validation.Constraint{}
 }
 
+// Migrate migrates the state to the current version.
+// This is mostly done to pass the validation of the current version.
+// The infrastructure will be overwritten by the terraform outputs after the validation.
+func (s *State) Migrate() error {
+	// In v2.13.0 the ClusterEndpoint and InClusterEndpoint fields were added.
+	// So they are expected to be empty when upgrading to this version.
+	if semver.Compare(constants.BinaryVersion().String(), "v2.13.0") != -1 {
```

Comment:
should we add a todo to remove this code in `v2.14`? this would be "wrong" in v2.15

Reply:
Yes we should add a TODO. Also I'll add a upper bound to the version check. Then we are "safe" either way. (We are already since the only set the values if they are empty).